tsdb/block.go

```diff
@@ -479,6 +487,12 @@ func (r blockIndexReader) LabelValueFor(id uint64, label string) (string, error)
 	return r.ir.LabelValueFor(id, label)
 }
 
+// LabelNamesFor returns all the label names for the series referred to by IDs.
+// The names returned are sorted.
+func (r blockIndexReader) LabelNamesFor(ids ...uint64) ([]string, error) {
```

Comment:
Needing to implement this here feels cumbersome to me, I feel that the interface we're implementing it shouldn't force us to do that, i.e. it should be split into two. But I don't have enough knowledge of the packet yet to have a better picture.